addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -18,10 +18,230 @@ var globalSelector = {
     is: () => false,
 };
 
+/**
+ * Mixin that provides smooth scroll behaviour.
+ *
+ * Call setDraggable in order to set the behaviour.
+ * Call unsetDraggable to ensure proper resource clearing.
+ */
+const SmoothOnDragMixin = {
+
+    /**
+     * Auto-scroll options.
+     *
+     * @private
+     */
+    _smoothOnDragOptions: {
+        $element: null,
+        $scrollTarget: null,
+        autoScrollHandler: null,
+        cursorAt: {
+            left: 0,
+            top: 0
+        },
+        draggableOffset: 0,
+        mainNavBarHeight: 0,
+        scrollOffsetThreshold: 150,
+        scrollStep: 20,
+        scrollStepDirection: 1,
+        scrollStepDirectionEnum: {
+            up: -1,
+            down: 1
+        },
+        scrollDecelerator: 0,
+        scrollTimer: 5,
+        visibleBottomOffset: 0,
+        visibleTopOffset: 0,
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Sets the element smoothly draggable.
+     * Do not forget to call unsetDraggable to ensure proper resource clearing.
+     * @see {@link unsetDraggable}.
+     *
+     * @public
+     * @param {jQuery} $element The element the smooth drag has to be set on.
+     * @param {Object} jQueryDraggableOptions The configuration to be passed to the jQuery draggable function.
+     */
+    setDraggable: function ($element, jQueryDraggableOptions) {
+        this._smoothOnDragOptions.$element = $element;
+        this._smoothOnDragOptions.$scrollTarget = $('html');
+        this._smoothOnDragOptions.mainNavBarHeight = $('.o_main_navbar').innerHeight();
+        this._smoothOnDragOptions.cursorAt = jQueryDraggableOptions.cursorAt || {left: 0, top: 0};
+        jQueryDraggableOptions.scroll = false;
+        var self = this;
+        let draggableOptions = Object.assign({}, jQueryDraggableOptions, {
+            start: function (ev, ui) {
+                self._onSmoothDragStart(
+                    ev,
+                    ui,
+                    jQueryDraggableOptions.start ?
+                        jQueryDraggableOptions.start.bind(this) : jQueryDraggableOptions.start
```

Comment:
I don't understand this, why passing the value when it is falsy?
Why not
```javascript
const callback = jQueryDraggableOptions.start ? jQueryDraggableOptions.start.bind(this) : null;
this._onSmoothDragStart(ev, ui, callback);
```
?
Or even:
```javascript
start: (ev, ui) => this._onSmoothDragStart(ev, ui, jQueryDraggableOption.start),
```
and then calling the function with the right `this` in `_onSmoothDragStart`